executor/insert_common.go

```diff
@@ -368,102 +352,115 @@ func (e *InsertValues) getColDefaultValue(idx int, col *table.Column) (d types.D
 	return defaultVal, nil
 }
 
-// initDefaultValues fills generated columns, auto_increment column and empty column.
+// fillColValue fills the column value if it is not set in the insert statement.
+func (e *InsertValues) fillColValue(datum types.Datum, idx int, column *table.Column, hasValue bool) (types.Datum,
+	error) {
+	if mysql.HasAutoIncrementFlag(column.Flag) {
+		d, err := e.adjustAutoIncrementDatum(datum, hasValue, column)
+		if err != nil {
+			return types.Datum{}, errors.Trace(err)
+		}
+		return d, nil
+	} else if !hasValue {
```

Comment:
s/else if/ if

Reply:
The "not auto increment" && "not has value case".

Reply:
If not auto increment and not has value, we could return the datum directly.

Reply:
else if can be changed to if, rest LGTM